packages/eslint-plugin/docs/rules/no-invalid-this.md

```diff
@@ -0,0 +1 @@
+# Disallow `this` keywords outside of classes or class-like objects (`no-invalid-this`)
```

Comment:
Example of an extension rule doc:

https://github.com/typescript-eslint/typescript-eslint/blob/1fd86befa6a940a0354c619dd2da08a5c5d69fb4/packages/eslint-plugin/docs/rules/comma-spacing.md

Key points:
- explain what we've added
- show how to turn it on
- link to the base rule docs.